src/ElmFormat/Render/Box.hs

```diff
@@ -1493,65 +1493,9 @@ formatExpression' elmVersion importInfo context aexpr =
                         ]
                     |> expressionParens AmbiguousEnd context -- TODO: not tested
 
-        AST.Expression.Case (subject,multiline) clauses ->
-            let
-                opening =
-                  case
-                    ( multiline
-                    , formatCommentedExpression elmVersion importInfo SyntaxSeparated subject
-                    )
-                  of
-                      (False, SingleLine subject') ->
-                          line $ row
-                              [ keyword "case"
-                              , space
-                              , subject'
-                              , space
-                              , keyword "of"
-                              ]
-                      (_, subject') ->
-                          stack1
-                              [ line $ keyword "case"
-                              , indent subject'
-                              , line $ keyword "of"
-                              ]
-
-                clause (pat, expr) =
-                    case
-                      ( pat
-                      , (formatPattern elmVersion False $ (\(AST.Commented _ x _) -> x) pat)
-                          |> negativeCasePatternWorkaround pat
-                      , formatCommentedStack (formatPattern elmVersion False) pat
-                          |> negativeCasePatternWorkaround pat
-                      , formatHeadCommentedStack (formatExpression elmVersion importInfo SyntaxSeparated) expr
```

Comment:
Performance should be slightly better because we don't format twice the pattern anymore.